src/winter_onboarding_2021/fleet_management_service/handlers/user.clj

```diff
@@ -1,38 +1,98 @@
 (ns winter-onboarding-2021.fleet-management-service.handlers.user
   (:require [ring.util.response :as response]
+            [next.jdbc :as jdbc]
+            [clojure.data.json :as json]
             [clojure.spec.alpha :as s]
             [crypto.password.bcrypt :as password]
+            [clj-http.client :as client]
             [hiccup.page :refer [html5]]
+            [clj-time.coerce :as c]
+            [clj-time.core :as t]
             [winter-onboarding-2021.fleet-management-service.session :as session]
             [winter-onboarding-2021.fleet-management-service.views.user :as view]
             [winter-onboarding-2021.fleet-management-service.models.user :as user-model]
             [winter-onboarding-2021.fleet-management-service.specs :as specs]
             [winter-onboarding-2021.fleet-management-service.views.layout :as layout]
-            [winter-onboarding-2021.fleet-management-service.utils :as utils]))
+            [winter-onboarding-2021.fleet-management-service.utils :as utils]
+            [winter-onboarding-2021.fleet-management-service.config :as config]
+            [winter-onboarding-2021.fleet-management-service.models.invite :as invite]
+            [winter-onboarding-2021.fleet-management-service.db.core :as db-core]))
 
 (defn signup-form [req]
-  (let [user (:user req)]
+  (let [user (:user req)
+        token (get-in req [:params :token])]
     (if (nil? user)
       (response/response (html5 (layout/application
                                  req
                                  "Sign up"
-                                 (view/signup-form))))
+                                 (view/signup-form token))))
       (response/redirect "/users/dashboard"))))
 
-(defn user-exist? [email]
+(defn- user-exist? [email]
   (not-empty (user-model/find-by-keys {:users/email email})))
 
+(defn- recaptcha-invalid? [g-recaptcha-response]
+  (try (let [res (client/post "https://www.google.com/recaptcha/api/siteverify"
```

Comment:
Should we add this hardcoeded URL in some config or constants file?

Reply:
Used only once, but still we can consider.